### ISSUE TITLE
True version string in speller test result output (Bugzilla Bug 491)

### ISSUE THREAD

**Issue by albbas** [giellalt/giella-core#253]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 491
Date: 2007-08-28T08:03:22+02:00
From: @snomos (Originally created by @snomos)
To:   Saara Huhmarniemi &lt;saara.huhmarniemi&gt;
CC:   @albbas

Depends on:   #492
Last updated: 2007-08-30T13:15:58+02:00

**Comment by albbas** [comment]
## Comment 1745
Date: 2007-08-28 08:03:22 +0200
From: @snomos

Presently, the version string is almost useless, unless you edit it manually. I may typically look like this:

public

which isn't very informative. This can certainly be improved, and the most informative solution would be to extract the version info provided by the speller lexicon. It would go like this:

1. when extracting the test data into a word list to send through the speller, prepend/add the entry 'nuvviD' as the first line, possibly with the correction to 'Divvun'.
2. when reading & processing the test results, treat this first line specially:
- look for a suggestion containing commas or spaces (or the sequence ', ' - comma + space)
- use that suggestion as the version string (it should be something like 'Julevsáme, public beta 2, 2007-08-23')
- discard the first error/correction/suggestions record alltogether - it should not be part of the final test result output
- if no such suggestion is found, leave the version string given as command line parameter as is, as a fallback mechanism in case something has gone wrong with the speller lexicon signature creation.

While I should I have no problems doing 1, I would need help with the second step.

**Comment by albbas** [comment]
## Comment 1757
Date: 2007-08-28 13:09:59 +0200
From: Saara Huhmarniemi &lt;saara.huhmarniemi&gt;

Sorry, I will not mark this as fixed yet.

**Comment by albbas** [comment]
## Comment 1774
Date: 2007-08-30 09:57:47 +0200
From: @snomos

*** Bug #446 has been marked as a duplicate of this bug. ***

**Comment by albbas** [comment]
## Comment 1776
Date: 2007-08-30 10:08:44 +0200
From: @snomos

It looks very good, but it doesn't work yet due to bug #492.

**Comment by albbas** [comment]
## Comment 1791
Date: 2007-08-30 13:15:58 +0200
From: @snomos

It works now, thanks a lot:)

Bugreport closed.
